python/astro_metadata_translator/translators/sdss.py

```diff
@@ -0,0 +1,226 @@
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the header is recognized by this class. `False`
+            otherwise.
+        """
+        if (cls.is_keyword_defined(header, "ORIGIN") and cls.is_keyword_defined(header, "CCDMODE") and
+                "SDSS" in header["ORIGIN"] and "DRIFT" in header["CCDMODE"]):
+            return True
+        return False
+
+    @cache_translation
+    def to_detector_unique_name(self):
+        # Docstring will be inherited. Property defined in properties.py
+        if self.is_keyword_defined(self._header, "CAMCOL"):
+            return self.to_physical_filter()+str(self._header["CAMCOL"])
+        else:
+            raise ValueError("CAMCOL key is not definded")
+
+    @cache_translation
+    def to_detector_num(self):
+        # Docstring will be inherited. Property defined in properties.py
+        return self.detector_name_id_map[self.to_detector_unique_name()]
+
+    @cache_translation
+    def to_observation_id(self):
+        """Calculate the observation ID.
+
+        Returns
+        -------
+        observation_id : `str`
+            A string uniquely describing the observation.
+            This incorporates the run, camcol, filter and frame.
+        """
+        return " ".join([str(self._header[el]) for el in
+                        ["RUN", "CAMCOL", "FILTER", "FRAME"]])
+
+    @cache_translation
+    def to_datetime_begin(self):
+        # Docstring will be inherited. Property defined in properties.py
+        # We know it is UTC
+        value = self._from_fits_date_string(self._header["DATE-OBS"],
+                                            time_str=self._header["TAIHMS"], scale="tai")
+        self._used_these_cards("DATE-OBS", "TAIHMS")
+        return value
+
+    @cache_translation
+    def to_datetime_end(self):
+        # Docstring will be inherited. Property defined in properties.py
+        return self.to_datetime_begin() + self.to_exposure_time()
+
+    @cache_translation
+    def to_location(self):
+        """Calculate the observatory location.
+
+        Returns
+        -------
+        location : `astropy.coordinates.EarthLocation`
+            An object representing the location of the telescope.
+        """
+
+        # Look up the value since files do not have location
+        value = EarthLocation.of_site("apo")
+
+        return value
+
+    @cache_translation
+    def to_observation_type(self):
+        """Calculate the observation type.
+
+        Returns
+        -------
+        typ : `str`
+            Observation type. Normalized to standard set.
+        """
+        obstype_key = "FLAVOR"
+        if not self.is_key_ok(obstype_key):
+            return "none"
+        obstype = self._header[obstype_key].strip().lower()
+        self._used_these_cards(obstype_key)
+        return obstype
```

Comment:
How does this ever return "science"?

Reply:
The FLAVOR card contains the string "science" for science observations.